java/vector/src/main/java/org/apache/arrow/vector/ipc/message/MessageChannelResult.java

```diff
@@ -0,0 +1,104 @@
+/**
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.apache.arrow.vector.ipc.message;
+
+import java.nio.ByteBuffer;
+
+import org.apache.arrow.flatbuf.Message;
+
+/**
+* Class to hold resulting Message and message information when reading messages from a ReadChannel.
+*/
+public class MessageChannelResult {
+
+  /**
+  * Construct a container to hold a message result.
+  *
+  * @param messageLength the length of the message read in bytes
+  * @param messageBuffer contains the raw bytes of the message
```

Comment:
I think you should be clear here that the "message" is the message _metadata_, and does not include the body (this can be inferred by reading onward of course...). Flatbuffers uses `ByteBuffer` as it binary interface is my understanding (I have confirmed this by looking at the .java files generated by flatc), and so this will help prevent possible confusion why there is ByteBuffer here instead of ArrowBuf